src/transport_ofi.h

```diff
@@ -871,13 +872,22 @@ shmem_transport_fetch_atomic(void *target, const void *source, void *dest,
 
 
 /* detect atomic limitation on the fly and provide software reduction support
-    if needed */
+    if needed, if supported update fragmentation size accordingly */
 static inline
 int shmem_transport_atomic_supported(shm_internal_op_t op,
                                      shm_internal_datatype_t datatype)
 {
    size_t size = 0;
    int ret = fi_atomicvalid(shmem_transport_ofi_epfd, datatype, op, &size);
+
+   if(size > 0) {
+      shmem_transport_ofi_max_atomic_size = size * SHMEM_Dtsize[datatype];
```

Comment:
Yikes!  The global state here will be a headache when we add threading support.  I'd rather update the transport layer API to pass the right info than stash it in library state.